TwoNote/app/src/main/java/com/microsoft/device/display/samples/twonote/NoteSelectionListener.kt

```diff
@@ -1,15 +1,40 @@
 package com.microsoft.device.display.samples.twonote
 
+import android.content.Context
+import android.util.Log
 import android.view.ActionMode
 import android.view.Menu
 import android.view.MenuItem
 import android.widget.AbsListView
+import android.widget.ArrayAdapter
 import android.widget.ListView
+import androidx.core.view.get
+import androidx.core.view.iterator
+import androidx.core.view.size
+import com.microsoft.device.display.samples.twonote.model.INode
+import java.io.File
 
-class NoteSelectionListener(private var host: NoteListFragment, private var listView: ListView) : AbsListView.MultiChoiceModeListener {
+class NoteSelectionListener(private var host: NoteListFragment, private var listView: ListView,
+                            private var context: Context, private var arrayAdapter: ArrayAdapter<INode>?) : AbsListView.MultiChoiceModeListener {
 
     override fun onActionItemClicked(mode: ActionMode, item: MenuItem): Boolean {
-        // TODO: add action handling here
+        when (item.itemId) {
+            R.id.action_delete -> {
+                val fileHandler = FileHandler()
+                arrayAdapter?.let {
+                    val list = listView.checkedItemPositions
+                    for (i in it.count - 1 downTo 0) {
+                        if (list.get(i)) {
+                            it.getItem(i)?.let {inode ->
+                                fileHandler.delete(context, "", inode)
+                            }
+                        }
+                    }
+                    arrayAdapter?.notifyDataSetChanged()
+                }
+                onDestroyActionMode(mode)
```

Comment:
I have not done a ton of research but idk if you actually have to call this method, or if it gets called automatically? I guess this is just a question haha like did it work without this explicit call

Reply:
deleting the file doesn't cause a deselection event, so the toolbar doesn't change or disappear. there might be a better way to do this, but right now forcing it to close after deleting works